src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.cs

```diff
@@ -299,6 +301,11 @@ private string GetInstanceId(IDictionary<string, StringValues> header)
                 return instanceId;
             }
             return string.Empty;
-        }
+        }
+
+        protected override Task ClientConnectionsCompletedTask()
+        {
+            return Task.WhenAll(_clientConnectionManager.ClientConnections.Values.Select(c => c.CompleteTask));

```

Comment:
>_clientConnectionManager [](start = 32, length = 24)

this is for connections clients in this app server not for this service connection, `_connectionIds` lives connections for this service connection

Reply:
So it's a global manager? Singleton?

Well, seems like it's an inherent barrier we've talked in the previous PR.